src/backend/main/py.main/caliopen_main/discussion/core/discussion.py

```
@@ -117,6 +117,20 @@ def get(self, user, min_il, max_il, limit, offset):
         responses = self.build_responses(user, discussions)
         return {'discussions': list(responses), 'total': total}
 
+    def get_one(self, user, discussion_id, min_il=0, max_il=100):
+
+        discussion_index = DIM(user).get_by_id(discussion_id, min_il, max_il)
+        if not discussion_index:
+            raise NotFound
+        try:
+            discussion_core = Discussion.get(user, discussion_index.discussion_id)
+        except NotFound:
```

let the exception raise naturally not except E; raise E